packages/test/test-end-to-end-tests/src/test/gc/gcStateResetInSummaries.spec.ts

```diff
@@ -228,23 +213,52 @@ describeFullCompat("GC state reset in summaries", (getTestObjectProvider) => {
         const newDataStore = await dataObjectFactory.createInstance(mainDataStore.containerRuntime);
         mainDataStore._root.set("newDataStore", newDataStore.handle);
 
-        // Validate that GC ran.
-        await validateGCRan(summarizerClient);
-
-        // Mark the data store as unreferenced by deleting its handle from the DDS. Validate that GC ran and the
-        // unreferenced data store is marked as such in GC state.
+        // Mark the data store as unreferenced by deleting its handle from the DDS.
         mainDataStore._root.delete("newDataStore");
-        await validateGCRan(summarizerClient, newDataStore.id);
 
-        // Load a new summarizer from the last summary with GC disabled. Validate that GC did not run and that
-        // previous GC state is removed expect for used routes which marks all data stores as referenced.
+        // Validate that GC ran and the unreferenced data store is marked as such in GC state.
+        await validateGCState(
+            summarizerClient,
+            true /* shouldGCRun */,
+            false /* shouldRegenerateSummary */,
+            [newDataStore.id],
+        );
+
+        // Load a new summarizer from the last summary with GC disabled.
         assert(latestAckedSummary !== undefined, "Summary ack isn't available as expected");
         const summarizerClient2 = await getNewSummarizer(
             true /* disableGC */,
             undefined /* gcAllowed */,
             latestAckedSummary.summaryAck.contents.handle,
         );
-        await validateGCDidNotRun(summarizerClient2);
+        // Validate that GC does not run and the summary is regenerated because GC was disabled.
+        await validateGCState(
+            summarizerClient2,
+            false /* shouldGCRun */,
+            true /* shouldRegenerateSummary */,
+        );
+
+        // Validate that GC does not run and the summary should be regenerated again. The summary is regenerated only
```

Comment:
Should this read, "summary should not be regenerated"?

Reply:
yes, you are right. copy paste error :)